src/libcore/iter/range.rs

```diff
@@ -95,11 +95,13 @@ macro_rules! step_impl_unsigned {
             }
 
             #[inline]
+            #[rustc_inherit_overflow_checks]
             fn add_one(&self) -> Self {
                 *self + 1
```

Comment:
`Add::add(*self, 1)` is what @alexcrichton meant. Similar situation on the rest.